### DISCUSSION TITLE
MSB und LSB ?

### DISCUSSION THREAD

**Discussion by Kenaschon** [dewenni/ESP32-Jarolift-Controller#13] [General]
Hallo zusammen,

was muß ich bei MSB, LSB, Seriennummer und Device Counter eingeben?

**Comment by dewenni** [comment]
hattest du das Projekt von Madmartin nicht auch schon laufen? Ich hatte da in den Issues einen Betrag von dir gesehen.
Bezüglich MSB, LSB, Seriennummer und Device Counter verhält sich dieses Projekt hier gleich.

siehe auch: [Migration](https://github.com/dewenni/ESP32-Jarolift-Controller?tab=readme-ov-file#migration)

**Comment by Kenaschon** [comment]
Habe jetzt einen neuen ESP32 genommen und ganz neu und frisch geflasht.

Nun geht es. Event. hatte der andere ESP einen hau.

Vielen Dank für die tolle Umsetzung.

**Reply by dewenni** [reply]
ok, prima. Konntest du auch deine bestehende Seriennummer verwenden und musstest nichts neu anlernen?

**Comment by Kenaschon** [comment]
Ich konnte die Werte 1:1 übernehmen. 

Bei Madmartin sind die Werte so"0x2719xxxx" und hier eben nur "2719xxxx".

Das ist der einzige Unterschied.

Warum wird eigentlich so ein Geheimnis um diese Werte gemacht?

**Comment by markus-hi** [comment]
Hallo,

ich muss das Thema nochmal anwärmen: da ich nicht von einem bestehendem Projekt migriere kann ich diese Werte nicht übernemen. Aber so habe ich überhaupt keinen Schimmer, was hier für Werte rein müssen?!? Kann mir mal jemand helfen bitte?
Die Seriennummer ist anscheinend frei wählbar, aber MSB und LSB auch?

Danke!

**Reply by dewenni** [reply]
nein, der MSB und LSB ist ein "Geheimnis" des Herstellers und daher auch nicht Bestandteil der Software.
Aber google mal nach "jarolift master msb" und dann bei Pastebin ;-)

**Comment by markus-hi** [comment]
Ah okay, deswegen die Geheimnistuerei 🤣😉

Danke für den Tip, damit komm ich klar 👍

**Reply by bart83-tech** [reply]
Hi, hast du es gechafft? Ich habe die Codes gefunden? Wie muss ich diese eintragen? Die 0x muss ich wahrscheinlich weglassen. 
Meine MSB endet mit: XXXX3a9b
und die LSB endet mit: XXXX3e0c

Die Seriennummer kann beliebieg sein oder? Ich bekomme momentan das einlernen nicht hin. Viele Grüße

**Reply by markus-hi** [reply]
Hi, ja mittlerweile läuft es bei mir. Ja du musst das 0x weglassen. Der LSB endet bei mir aber anders. Die Seriennummer kann beliebig sein, muss aber durch 16 teilbar sein hab ich wo gelesen. Hier steht viel zu dem Thema:
https://www.mikrocontroller.net/topic/471508

Gruß - Markus

**Reply by bart83-tech** [reply]
Hi, alles klar. Der LSB war auch XXXX0835. Ich hatte eine falsche drin.
Dann probiere ich das mal aus.

**Reply by markus-hi** [reply]
Ja das sieht jetzt korrekt aus, so endet die bei mir auch.

Echt geiles Projekt 👍👏 Ich werd im Sommer die restlichen Rollläden noch umbauen bzw. aufrüsten mir jarolifts.

**Reply by bart83-tech** [reply]
Vielen Dank. Jetzt funktioniert alles. Ja, es ist ein wirklich klasse Projekt mit toller Umsetzung.
